server/server.go

```diff
@@ -107,110 +126,95 @@ func (serv *shipperServer) PublishEvents(_ context.Context, req *messages.Publis
 	if req.Uuid != "" && req.Uuid != serv.uuid {
 		resp.AcceptedIndex = serv.GetAcceptedIndex()
 		resp.PersistedIndex = serv.GetPersistedIndex()
-		serv.logger.
-			With(
-				"expected", serv.uuid,
-				"actual", req.Uuid,
-			).
-			Debugf("shipper UUID does not match, all events rejected")
+		serv.logger.Debugf("shipper UUID does not match, all events rejected. Expected = %s, actual = %s", serv.uuid, req.Uuid)
 
 		return resp, nil
 	}
 
 	for _, e := range req.Events {
-		err := serv.queue.Publish(e)
+		_, err := serv.publisher.Publish(e)
 		if err == nil {
-			resp.AcceptedIndex++
+			resp.AcceptedCount++
 			continue
 		}
 
-		log := serv.logger.
-			With(
-				"event_count", len(req.Events),
-				"accepted_count", resp.AcceptedCount,
-			)
-
 		if errors.Is(err, queue.ErrQueueIsFull) {
-			log.Warnf("queue is full, not all events accepted")
+			serv.logger.Debugf("queue is full, not all events accepted. Events = %d, accepted = %d", len(req.Events), resp.AcceptedCount)
 		} else {
-			err = fmt.Errorf("failed to enqueue an event: %w", err)
+			err = fmt.Errorf("failed to enqueue an event. Events = %d, accepted = %d: %w", len(req.Events), resp.AcceptedCount, err)
 			serv.logger.Error(err)
 		}
 
 		break
 	}
 
-	// it is cheaper to check than always using atomic
-	if resp.AcceptedCount > 0 {
-		atomic.AddInt64(&serv.acceptedIndex, int64(resp.AcceptedCount))
-	}
 	resp.AcceptedIndex = serv.GetAcceptedIndex()
 	resp.PersistedIndex = serv.GetPersistedIndex()
 
 	serv.logger.
-		With(
-			"event_count", len(req.Events),
-			"accepted_count", resp.AcceptedCount,
-			"accepted_index", resp.AcceptedIndex,
-			"persisted_index", resp.PersistedIndex,
-		).
-		Debugf("finished publishing a batch")
+		Debugf("finished publishing a batch. Events = %d, accepted = %d, accepted index = %d, persisted index = %d",
+			len(req.Events),
+			resp.AcceptedCount,
+			resp.AcceptedIndex,
+			resp.PersistedIndex,
+		)
 
 	return resp, nil
 }
 
 // PublishEvents is the server implementation of the gRPC PersistedIndex call.
 func (serv *shipperServer) PersistedIndex(req *messages.PersistedIndexRequest, producer pb.Producer_PersistedIndexServer) error {
-	// first we reply with current values and then subscribe to notifications and start streaming
-	err := producer.Send(&messages.PersistedIndexReply{
-		Uuid:           serv.uuid,
-		PersistedIndex: serv.GetPersistedIndex(),
-	})
-	if err != nil {
-		return err
-	}
-
-	serv.logger.Debug("subscribing client for persisted index change notification...")
-	ch, stop, err := serv.notifications.subscribe()
-	if err != nil {
-		return err
+	atomic.AddInt64(&serv.indexSubscribers, 1)
+	defer atomic.AddInt64(&serv.indexSubscribers, -1)
+
+	p, _ := peer.FromContext(producer.Context())
+	addr := addressString(p)
+	serv.logger.Debugf("%s subscribed for persisted index change", addr)
```

Comment:
Is this useful if the shipper and inputs are always running on the same host? Does it still work if we use a Unix domain socket instead of TCP?

Reply:
I guess we'll find out, so you think it's not a good idea to identify a client for debugging?

Reply:
It is a good idea, but I'm not sure the address will be unique per client. Likely what you want is the input or stream ID.

Reply:
Then we would need to make Input ID and Stream ID a part of the `messages.PersistedIndexRequest`, not sure we want that just for debugging. I'd rather have IP rather than nothing.

Reply:
I'll remove logging the address, since the feedback is negative.